arkeocli/claim.go

```
@@ -0,0 +1,142 @@
+		Args:  cobra.ExactArgs(0),
+		RunE:  runClaimCmd,
+	}
+
+	flags.AddTxFlagsToCmd(claimCmd)
+	claimCmd.Flags().String("provider-pubkey", "", "provider pubkey")
+	claimCmd.Flags().String("contract-id", "", "id of contract")
+	claimCmd.Flags().String("service", "", "service name")
+	claimCmd.Flags().Int64("nonce", 0, "requests claimed (must increment each call)")
+	return claimCmd
+}
+
+func runClaimCmd(cmd *cobra.Command, args []string) (err error) {
+	clientCtx, err := client.GetClientTxContext(cmd)
+	if err != nil {
+		return err
+	}
+	nonce, _ := cmd.Flags().GetInt64("nonce")
+	if nonce == 0 {
+		nonceString, err := promptForArg(cmd, "Specify nonce: ")
+		if err != nil {
+			return err
+		}
+		nonce, err = strconv.ParseInt(nonceString, 10, 64)
+		if err != nil {
+			return err
+		}
+	}
+	key, err := ensureKeys(cmd)
+	if err != nil {
+		return err
+	}
+	addr, err := key.GetAddress()
+	if err != nil {
+		return
+	}
+	creator := addr.String()
+
+	clientCtx = clientCtx.WithFromName(key.Name).WithFromAddress(addr)
+	if err = client.SetCmdClientContext(cmd, clientCtx); err != nil {
+		return
+	}
+
+	spenderPubkeyStr, err := toPubkey(cmd, addr)
+	if err != nil {
+		return
+	}
+
+	spenderPubkey, err := common.NewPubKey(spenderPubkeyStr)
+	if err != nil {
+		return err
+	}
+
+	queryCtx, err := client.GetClientQueryContext(cmd)
+	if err != nil {
+		return err
+	}
+
+	queryClient := types.NewQueryClient(queryCtx)
+
+	var contract types.Contract
+	contractID, _ := cmd.Flags().GetUint64("contract-id")
+	if contractID != 0 {
+		params := &types.QueryFetchContractRequest{ContractId: contractID}
+		res, err := queryClient.FetchContract(cmd.Context(), params)
+		if err != nil {
+			return errors.Wrapf(err, "error fetching contract %d", contractID)
+		}
+		if res == nil {
+			return fmt.Errorf("no contract found %d", contractID)
+		}
+		contract = res.GetContract()
+	} else {
+		providerPubkey, _ := cmd.Flags().GetString("provider-pubkey")
```

please don't ignore this error